### PR TITLE
Cleaned up Sylius dependencies

### PR DESCRIPTION
**Pull request by NeverResponse**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

* Removed unused or duplicated entries
* Added missing dependencies
* Sorted requirements alphabetically with exclusion of php and phpspec
* Sylius dependencies order:
  * Components first
  * Bundles follow
* Aligned authors' parameters

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -33,6 +30,10 @@
         "doctrine/doctrine-migrations-bundle":  "~1.0",
         "doctrine/orm":                         ">=2.4.8,<2.5",
         "doctrine/phpcr-odm":                   "^1.2.2",
+        "ext-exif":                             "*",
+        "ext-fileinfo":                         "*",
+        "ext-gd":                               "*",
+        "ext-intl":                             "*",
```
Comment (pjedrzejewski) [inline composer.json]: These are php extensions not libraries, so I think they should on top.

File: composer.json [hunk 2]
```diff
@@ -79,17 +80,18 @@
...
-        "behat/mink-extension":                 "~2.0",
-        "behat/mink-browserkit-driver":         "~1.2",
-        "behat/mink-selenium2-driver":          "~1.2",
-        "behat/mink":                           "~1.6",
-        "coduo/php-matcher":                    "~1.0",
-        "phpspec/phpspec":                      "^2.4@alpha",
-        "phpunit/phpunit":                      "~4.1",
-        "lakion/mink-debug-extension":          "^1.0.1",
-        "akeneo/phpspec-skip-example-extension": "~1.2"
+        "phpspec/phpspec":                       "^2.4@alpha",
```
Comment (pjedrzejewski) [inline composer.json]: Why blank line after this?

**Comment by pamil** [comment]
Can we stop aligning versions in `composer.json`? If you add package through `composer require` it's very likely not to remember about aligning it. Moreover, if the package name is the new longest one, all the previous package lines need changes.

**Comment by michalmarcinkowski** [comment]
Agree with @pamil about aligning versions. It makes maintenance painfull.

File: src/Sylius/Bundle/ApiBundle/composer.json [hunk 3]
```diff
@@ -20,18 +20,19 @@
...
     ],
     "require": {
-        "php":                      ">=5.5.9",
+        "php":                                  ">=5.5.9",
 
+        "friendsofsymfony/oauth-server-bundle": "~1.4",
         "sylius/resource-bundle":               "0.16.*",
-        "symfony/framework-bundle":             "~2.3",
-        "friendsofsymfony/oauth-server-bundle": "~1.4"
+        "symfony/framework-bundle":             "~2.3"
```
Comment (TheMadeleine) [inline src/Sylius/Bundle/ApiBundle/composer.json]: https://github.com/Sylius/Sylius/pull/3449/files update to symfony 2.7.5 has not been merged yet :)

**Comment by pjedrzejewski** [comment]
Thank you Jasiek!

**Pull request merged**